SIS_import/sis_upload.rb

```diff
@@ -58,12 +59,32 @@
 # the path of Canvas API call
 API_PATH="/api/v1/"
 
+
+# variables for Canva API call throttling
+@Canvas_call_count = 0
+@Canvas_start_time = Time.now
+@Canvas_time_interval_in_seconds = 3600
+@Canvas_end_time = @Canvas_start_time + @Canvas_time_interval_in_seconds # one minute apart
+@Canvas_allowed_call_number_during_interval = 3000
+
+# make sure the Canvas call is within API usage quota
+def Canvas_API_CALL_check
+	call_hash = sleep_according_to_timer_and_api_call_limit(@Canvas_start_time, @Canvas_end_time, @Canvas_call_count, @Canvas_time_interval_in_seconds, @Canvas_allowed_call_number_during_interval)
```

Comment:
Done. Now call_hash is the individual variable, with keys to "start_time", "end_time", "allowed_call_number_during_interval", etc.